applications/tari_dan_app_utilities/src/template_manager/implementation/manager.rs

```diff
@@ -190,6 +241,28 @@ impl<TAddr: NodeAddressable> TemplateManager<TAddr> {
         Ok(templates)
     }
 
+    pub fn add_pending_template(
+        &self,
+        template_address: tari_engine_types::TemplateAddress,
+        epoch: Epoch,
+    ) -> Result<(), TemplateManagerError> {
+        let template = DbTemplate::empty_pending(template_address, epoch);
+
+        let mut tx = self.global_db.create_transaction()?;
+        let mut templates_db = self.global_db.templates(&mut tx);
+        match templates_db.get_template(&template.template_address)? {
```

Comment:
Couldn't you use `template_exists` here?
```suggestion
        match templates_db.template_exists(&template.template_address)? {
```